src/Protocol/ActivityPub.php

```diff
@@ -90,6 +92,7 @@ public static function isRequest()
 	public static function fetchContent($url, $uid = 0)
 	{
 		if (empty($url)) {
+			Logger::info('Empty URL fetched', ['uid' => $uid, 'callstack' => System::callstack()]);
```

Comment:
I would prefer to set the `$url` parameter type-hint to `string` and fix the wrong calling code instead. Free exception logging, no need to add this check.

Reply:
Yes .. That's what I already started. But I followed the cascading path back and found that adding type-hints at every parameter call, where "url" could be null, it leads to a lot of changed code.

I'll add it in an extra PR and revert it for this PR ;-)

Reply:
What about a "huge" PR in the next dev-cycle:

"Adding type-hinting for each parameter, where we know the type based on the comment"?

I guess that this will create a temporary outbreak of PHP Fatals ;-) But we'll catch them more efficient than guessing for each new type-hint, which calls could lead to a wrong/null type and if this is the "end" of the execution path or just in the middle ...

Reply:
I like this scorched earth policy.